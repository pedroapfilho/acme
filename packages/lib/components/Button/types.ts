import { HTMLProps } from "react";

interface ButtonProps extends Omit<HTMLProps<HTMLButtonElement>, "size" | "as"> {
  fullWidth?: boolean;
}

export type { ButtonProps };
