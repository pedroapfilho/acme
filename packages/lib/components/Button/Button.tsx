import React, { FC } from "react";

const Button: FC<any> = ({ children }) => <button>Button: {children}</button>;

export default Button;
