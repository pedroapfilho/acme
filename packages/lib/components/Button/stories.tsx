import React, { FC } from "react";

import Button from "./";

export default {
	component: Button,
	title: "Button",
};

export const Default: FC = () => <Button>Something</Button>;
