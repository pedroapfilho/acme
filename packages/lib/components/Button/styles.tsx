import React from "react";
import styled, { css } from "styled-components";
import { ButtonProps } from "./types";

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const StyledButton = styled(({ size, fullWidth, color, ...props }) => <button {...props} />)<
  ButtonProps
>`
  cursor: pointer;

  &:focus {
    outline: none;
  }

  padding: ${(props) => props.theme.space[3]}px;

  border: none;

  background-color: ${(props) => props.theme.colors.background};

  color: ${(props) => props.theme.colors.text};

  ${(props) =>
    props.fullWidth &&
    css`
      width: 100%;
    `};
`;

export { StyledButton };
