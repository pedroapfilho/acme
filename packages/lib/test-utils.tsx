import React, { ReactElement } from "react";

// import theme from "./theme";

import { render, RenderOptions, RenderResult } from "@testing-library/react";

const Providers = ({ children }) => {
  return <>{children}</>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "queries">): RenderResult =>
  render(ui, { wrapper: Providers, ...options });

export * from "@testing-library/react";

export { customRender as render };
