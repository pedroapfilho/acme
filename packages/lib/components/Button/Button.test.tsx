import React from "react";
import { render } from "../../test-utils";
import Button from ".";

describe("Button", () => {
  it("Should match snapshots", () => {
    const { container } = render(<Button>Sample Text</Button>);

    expect(container).toMatchSnapshot();
  });

  it("Should render with no errors", () => {
    const { container } = render(<Button>Sample Text</Button>);

    expect(container).toBeInTheDocument();
  });
});
