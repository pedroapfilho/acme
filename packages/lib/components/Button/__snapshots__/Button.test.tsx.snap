// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match snapshots 1`] = `
<div>
  <button>
    Sample Text
  </button>
</div>
`;
